cmd/api/main.go

```
@@ -11,11 +12,29 @@ var Version = "dev"
 
 func main() {
 	if err := engine.Run(
-		os.Getenv,
+		readEnvOrFile,
 		os.Stdout,
 		Version,
 	); err != nil {
 		fmt.Fprintf(os.Stderr, "%s\n", err)
 		os.Exit(1)
 	}
 }
+
+func readEnvOrFile(envName string) string {
+	envContent := os.Getenv(envName)
+
+	if envContent == "" {
+		filename := os.Getenv(envName + "_FILE")
+
+		if filename != "" {
+			b, err := os.ReadFile(filename)
+
+			if err == nil {
```

If a user assigns a value to a _FILE variable, rather than silently failing, it would be better if the program failed quickly (with log.Fatal or something) to let the user know that something about their configuration is incorrect *before* starting the program, since starting the program acts as an assurance that their config is right (which, if err != nil, it is not).

---

Can I just call `panic()` with a descriptive message, or should I `log.Fatal`, and then `os.Exit(1)`? This is my first time writing go, so I don't know what is better

---

I've used `panic()` in the new commit

---

I see, it should be okay now